Source/Dafny/Resolver.cs

```diff
@@ -3303,15 +3306,15 @@ public class ModuleBindings
             if (nm == "Floor") {
               Object ee = GetAnyConst(m.Obj, consts);
               if (ee != null && m.Obj.Type.IsNumericBased(Type.NumericPersuation.Real)) {
-                BigInteger f, cc;
-                ((Basetypes.BigDec)ee).FloorCeiling(out f, out cc);
+                ((Basetypes.BigDec)ee).FloorCeiling(out var f, out _);
                 return f;
               }
             }
           }
         } else if (e is BinaryExpr bin) {
           Object e0 = GetAnyConst(bin.E0, consts);
           Object e1 = GetAnyConst(bin.E1, consts);
+          if (e0 == null || e1 == null) { return null; }
           bool isBool = bin.E0.Type == Type.Bool && bin.E1.Type == Type.Bool;
           bool isReal = bin.E0.Type.IsNumericBased(Type.NumericPersuation.Real)
                         && bin.E1.Type.IsNumericBased(Type.NumericPersuation.Real);
```

Comment:
Just like there is a distinction between `isInt` and `isInteger`, `isReal` also needs those two cases. In particular, comparisons between any numbers of the real persuasion is fine, but arithmetic operations should give up and return `null` for any `newtype`.

Reply:
Done